BCStudentSoftwareDevTeam/celts
# Interface graduation management

## Issue Description

Fixes #1315 

- The purpose of this issue is to create a page that allows celts admins to mark students as graduated in the data base.

- The page should include a way for the user to filter students by searching for individual users, by Bonner Cohort (and what academic year they are associated with), as well as students who are participating in the CCE minor (whether they have completed it or made any progress towards it. ).


## Changes
- Created a new html, js, logic, and controller file called graduationManagement.
- Created new functions for created setting the graduation status of a user and fetching all eligible users.
- Added a new field in the database `hasGraduated`
- Created a toggle that allows you to hide or show all graduated students.
- Stored all filter and toggle data in session state so that a page reload doesn't get rid of all your changes.

## Testing
- Checkout `interface-graduation-management`
- Head to the Graduation Management page in the Admin sidebar
- Boot up the page and make sure all students are seniors
- Mark students as graduated and refresh the page to make their state is permanent.
- Try to filter and toggle various buttons and make sure they properly filter the data as well as being properly stored in the session state. Make sure you can refresh and changes are maintained.
